Flutter/pdf-viewer/form-filling.md

```diff
@@ -0,0 +1,193 @@
+
+By default, the `PdfFlattenOption` will be `PdfFlattenOption.none`, which means the form fields (except signature fields) can be edited after save.  
+
+Refer to the following code example. 
+
+{% tabs %}
+{% highlight dart hl_lines="15 16" %}
+
+final PdfViewerController _pdfViewerController = PdfViewerController(); 
+ 
+@override 
+Widget build(BuildContext context) { 
+  return Scaffold( 
+    appBar: AppBar( 
+      title: const Text('Syncfusion Flutter PDF Viewer'), 
+      actions: <Widget>[ 
+        IconButton( 
+          icon: const Icon( 
+            Icons.save, 
+            color: Colors.white, 
+          ), 
+          onPressed: () async { 
+            final List<int> savedBytes = await _pdfViewerController 
+                .saveDocument(flattenOption: PdfFlattenOption.formFields); 
+          }, 
+        ), 
+      ], 
+    ), 
+    body: SfPdfViewer.asset( 
+      'assets/form_document.pdf', 
+      controller: _pdfViewerController, 
+    ), 
+  ); 
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Importing and Exporting form data 
+
+The Flutter PDF viewer allows the users to import and export form data to and from the PDF documents. The import and export of form data supports the following extensions. 
+
+* fdf 
+* xfdf 
+* json 
+* xml 
+
+The required file type can be chosen from the [DataFormat](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/DataFormat.html) enumeration. In the following sections only the `xfdf` file type is explained for brevity. 
+
+N> Import ‘package:syncfusion_flutter_pdf/pdf.dart’ in the Dart code to use the [DartFormat](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/DataFormat.html) parameter. 
+
+### Exporting form data 
+
+The `exportFormData` method exports the current data filled in the form fields into bytes list in the specified data format. Refer to the following code example.
+
+{% tabs %}
+{% highlight dart hl_lines="15 16" %}
+
+final PdfViewerController _pdfViewerController = PdfViewerController(); 
+ 
+@override 
+Widget build(BuildContext context) { 
+  return Scaffold( 
+    appBar: AppBar( 
+      title: const Text('Syncfusion Flutter PDF Viewer'), 
+      actions: <Widget>[ 
+        IconButton( 
+          icon: const Icon( 
+            Icons.outbox, 
+            color: Colors.white, 
+          ), 
+          onPressed: () async { 
+            final List<int> formDataBytes = _pdfViewerController 
+                .exportFormData(dataFormat: DataFormat.xfdf); 
+          }, 
+        ), 
+      ], 
+    ), 
+    body: SfPdfViewer.asset( 
+      'assets/form_document.pdf', 
+      controller: _pdfViewerController, 
+    ), 
+  ); 
+} 
+
+{% endhighlight %}
+{% endtabs %}
+
+N> When exporting, the signature form field data will not be exported.  
+ 
+### Importing form data 
+
+The `importFormData` method imports the data from a file of specified type and fills the saved data into the form fields. 
```

Comment:
The `importFormData` method imports the data from a file of a specified type and fills the saved data into the form fields.